src/functions-reference/hidden_markov_models.Rmd

```diff
@@ -0,0 +1,69 @@
+# Hidden Markov Models (with discrete hidden variables)
+
+An elementary first-order Hidden Markov model 
+is a probabilistic model over
+$N$ observations, $y_n$, and $N$ hidden states, $x_n$,
+which can be fully defined by the conditional distributions
+$p(y_n \mid x_n)$ and $p(x_n \mid x_{n - 1})$.
+When $x$ is continuous, the user can explicitly encode these distributions
+in Stan and use Markov chain Monte Carlo to integrate $x$ out.
+
+When each state $x$ takes a value over a discrete and finite set, say $\{1, 2, ..., K\}$,
+we can take advantage of the dependency structure
+to marginalize $x$ and compute $p(y)$.
+We start by defining the conditional observational distribution,
+stored in an $N \times K$ matrix $\omega$ with
```

Comment:
Is it NxK or KxN? The indexing is KxN.

Reply:
Also can the notation use k instead of i? Right now it's $i \in {1, ..., K}$ but I think $k \in {1, ..., K}$ is clearer.

Reply:
It's N x K. Each row is a simplex, and the probabilities are taken over the K possible states x_n can take.

Reply:
It's indexed $\omega_{k n}$ in the LaTeX below this is what I meant, sorry. That implies KxN.

Reply:
Also it looks like the Gamma notation is still `i`. I think stick with either `i` or `k`.

Reply:
My bad, it's K x N. Unfortunately, I get these things mixed up :(

Here's the [file](https://github.com/stan-dev/math/blob/develop/stan/math/prim/prob/hmm_marginal.hpp), see line 84 and 85. I'm updating the doc.